src/main/java/randoop/main/GenInputsAbstract.java

```diff
@@ -849,6 +849,16 @@ public enum InputSelectionMode {
   @Option("If true, use JUnit's reflective invocation; if false, use direct method calls")
   public static boolean junit_reflection_allowed = true;
 
+  /**
+   * If true, Randoop will add a definition of {@code assertArrayEquals(boolean[], boolean[])} to
+   * the generated JUnit tests and use it instead for checking the equality of boolean arrays. This
+   * option is a backward compatibility feature as {@code assertArrayEquals(boolean[], boolean[])}
+   * is not available in JUnit versions 4.11 and earlier.
+   */
+  @Option(
+      "If true, include an assertArrayEquals(boolean[], boolean[]) implementation in generated JUnit tests")
+  public static boolean legacy_boolean_array_check = false;
```

Comment:
I would give this a name like "junit_pre_4_12", since that explains the reason.
More importantly, throughout please handle all arrays, not just boolean ones.
If we encountered the problem with boolean arrays, then it is likely to arise for other types of arrays as well, and it will be easier for us to handle them all now than to discover and fix the problem later.

Reply:
It’s odd that boolean[] was the only type that was missing in assertArrayEquals, while all other primitive array types were supported prior to version 4.12. According to the release notes for version 4.12:

> Assert.assertArrayEquals() previously existed for all primitive array types, except boolean[]. This has now been added for boolean[]

Given this, should we consider adding support for other primitive arrays just in case?

Reply:
Thank you for clarifying.  I had assumed that all the array operations were added at the same time, uniformly.  Since only the boolean one is needed, your pull request is fine.